tachyon/math/base/big_int.h

```diff
@@ -436,6 +436,81 @@ struct ALIGNAS(internal::LimbsAlignment(N)) BigInt {
     return *this;
   }
 
+  constexpr BigInt operator&(const BigInt& other) const {
+    BigInt ret;
+    if constexpr (N == 1) {
+      ret[0] = limbs[0] & other[0];
+    } else if constexpr (N == 2) {
+      ret[0] = limbs[0] & other.limbs[0];
+      ret[1] = limbs[1] & other.limbs[1];
+    } else {
+      DoAnd(*this, other, ret);
+    }
+    return ret;
+  }
+
+  constexpr BigInt& operator&=(const BigInt& other) {
+    if constexpr (N == 1) {
+      limbs[0] &= other[0];
+    } else if constexpr (N == 2) {
+      limbs[0] &= other.limbs[0];
+      limbs[1] &= other.limbs[1];
+    } else {
+      DoAnd(*this, other, *this);
+    }
+    return *this;
+  }
+
+  constexpr BigInt operator|(const BigInt& other) const {
+    BigInt ret;
+    if constexpr (N == 1) {
+      ret[0] = limbs[0] | other[0];
+    } else if constexpr (N == 2) {
+      ret[0] = limbs[0] | other.limbs[0];
+      ret[1] = limbs[1] | other.limbs[1];
+    } else {
+      DoOr(*this, other, ret);
+    }
+    return ret;
```

Comment:
```suggestion
    return ret |= other;
```

Reply:
`ret` is an empty BigInt, so this is impossible.